First import

tiraboschi opened the pull request:
Verified on the positive flow on HE case.

tiraboschi commented:
@machacekondra @didib @irosenzw @sandrobonazzola Can you please start reviewing this?

tiraboschi commented on tasks/main.yml:
```diff
@@ -0,0 +1,104 @@
+        - "{{ he_hosts }}"
+
+    - name: Shutdown of HE hosts
+      command: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /etc/pki/ovirt-engine/keys/engine_id_rsa -p {{item.ssh.port}} -t root@{{item.address}} 'while pgrep qemu-kvm >/dev/null; do sleep 5; done; shutdown -h now'
```
This is not enough in HC case since the engine VM is running just on one of the three hosts

tiraboschi committed:
First import

tiraboschi committed:
Better check global maintenance mode

machacekondra commented on build.sh:
```diff
@@ -0,0 +1,53 @@
+  mkdir -p $PKG_DOC_DIR
+
+  # Create a symlink, so legacy role name does work:
+  ln -f -s $PKG_DATA_DIR_ORIG $ROLENAME_LEGACY 
```
You don't need this line.

machacekondra commented on build.sh:
```diff
@@ -0,0 +1,53 @@
+PKG_DATA_DIR=${PKG_DATA_DIR:-$ROLES_DATAROOT_DIR/$PACKAGE_NAME}
+PKG_DATA_DIR_ORIG=${PKG_DATA_DIR_ORIG:-$PKG_DATA_DIR}
+PKG_DOC_DIR=${PKG_DOC_DIR:-$DOC_DIR/$PACKAGE_NAME}
+ROLENAME_LEGACY="${ROLENAME_LEGACY:-$ROLES_DATAROOT_DIR/ovirt-shutdown-env}"
```
You don't need this line.

machacekondra commented on ovirt-ansible-shutdown-env.spec.in:
```diff
@@ -0,0 +1,54 @@
+%global rolename shutdown-env
+%global roleprefix oVirt.
+%global roleprefix_legacy ovirt-
```
not needed

machacekondra commented on ovirt-ansible-shutdown-env.spec.in:
```diff
@@ -0,0 +1,54 @@
+st = posix.stat(path)
+if st and st.type == "directory" then
+  os.execute('rm -rf "'..path..'"')
+end
```
22-29 can be revmoed

machacekondra commented on ovirt-ansible-shutdown-env.spec.in:
```diff
@@ -0,0 +1,54 @@
+export PKG_DATA_DIR_ORIG=%{_datadir}/%{ansible_roles_dir}/%{roleprefix}%{rolename}
+export PKG_DATA_DIR=%{buildroot}$PKG_DATA_DIR_ORIG
+export PKG_DOC_DIR=%{buildroot}%{_pkgdocdir}
+export ROLENAME_LEGACY=%{buildroot}%{_datadir}/%{ansible_roles_dir}/%{roleprefix_legacy}%{rolename}
```
can be removed

machacekondra commented on ovirt-ansible-shutdown-env.spec.in:
```diff
@@ -0,0 +1,54 @@
+
+%files
+%{_datadir}/%{ansible_roles_dir}/%{roleprefix}%{rolename}
+%{_datadir}/%{ansible_roles_dir}/%{roleprefix_legacy}%{rolename}
```
can be removed

tiraboschi committed:
Remove legacy naming

tiraboschi committed:
Warn about global mainteance mode if on HE

tiraboschi committed:
ignore errors on not HE hosts via ssh

ignore errors on not HE hosts via ssh to
avoid failing the whole playbook for unresponsive
hosts.

tiraboschi commented:
See https://bugzilla.redhat.com/show_bug.cgi?id=1578339#c15
